astropy/units/quantity.py

```diff
@@ -39,6 +39,15 @@
 _UNIT_NOT_INITIALISED = "(Unit not initialised)"
 _UFUNCS_FILTER_WARNINGS = {np.arcsin, np.arccos, np.arccosh, np.arctanh}
 
+def quantity_allclose(a, b, rtol=1.e-5, atol=None, **kwargs):
+    """
+    Returns True if two arrays are element-wise equal within a tolerance.
+
+    This is a :class:`~astropy.units.Quantity`-aware version of
```

Comment:
Now that we are making this a much more visible function, it should have a proper docstring, defining the arguments, etc.  See developer documentation (and other examples in code) for details on how this looks. This particular sentence would go under "Notes"

Reply:
That means `_unquantify_allclose_arguments` shall also have a proper docstring,right?

Reply:
It remains a private function and it is pretty obvious what it does, so I don't think it is essential, but feel free to add something short.